### ISSUE TITLE
false positive for 'import-self'

### ISSUE THREAD

**Issue by kain88-de** [PyCQA/pylint#2617]
### Steps to reproduce
1. checkout mdanalysis
2. install newest pylint
3. `cd mdanalysis/package && pylint MDAnalysis`

### Current behavior
Empty function with a pass statement fails the lint. See for [example](https://github.com/MDAnalysis/mdanalysis/blob/aaf2f9a5cd7a21a3f58cb1e7f9f71ffc71281404/package/MDAnalysis/lib/formats/__init__.py#L24)

Corresponding failing [travis job](https://travis-ci.org/MDAnalysis/mdanalysis/jobs/460318184#L719)
and issue https://github.com/MDAnalysis/mdanalysis/issues/2148

### Expected behavior

Code is fine

### pylint --version output
https://travis-ci.org/MDAnalysis/mdanalysis/jobs/460318184#L681

This only happens with the newest version. Older versions of pylint do not report an error.

**Comment by orbeckst** [comment]
> empty function with a pass statement fails the lint. 

I think you meant something like "standard relative import semantics `from . import some_mod` fail the lint", as show in the linked example.

**Comment by PCManticore** [comment]
Thanks for reporting the bug!

**Comment by ajasmin** [comment]
In a module named foo.threading, I have:
```
import threading
```

which in pylint 2.6.0 yields *import-self*

I believe that since python now uses absolute import, it should unambiguously import `threading` from the standard library and the diagnostic is incorrect.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas I cannot reproduce this. Might have been fixed!

**Comment by Pierre-Sassoulas** [comment]
Closing as the issue is pointing to code that evolved since and it has a duplicate.
